lib/ui/pages/news_list/news_list_page.dart

```diff
@@ -1,85 +1,63 @@
 import 'package:allthenews/ui/common/util/dimens.dart';
-import 'package:allthenews/ui/common/util/strings.dart';
-import 'package:allthenews/ui/common/widget/primary_text_button.dart';
-import 'package:allthenews/ui/pages/news_list/primary_news/primary_news_list_entity.dart';
-import 'package:allthenews/ui/pages/news_list/primary_news/primary_news_list_view.dart';
+import 'package:allthenews/ui/common/widget/primary_icon_button.dart';
 import 'package:allthenews/ui/pages/news_list/secondary_news/secondary_news_list_entity.dart';
-import 'package:allthenews/ui/pages/news_list/secondary_news/secondary_news_list_page.dart';
 import 'package:allthenews/ui/pages/news_list/secondary_news/secondary_news_list_view.dart';
 import 'package:flutter/material.dart';
 
 abstract class _Constants {
-  static const sectionHeaderPadding = 10.0;
+  static const sectionHeaderPadding = 16.0;
 }
 
-class NewsListPage extends StatefulWidget {
-  @override
-  _NewsPageState createState() => _NewsPageState();
-}
+class NewsListPage extends StatelessWidget {
+  final List<SecondaryNewsListEntity> listEntities;
+  final String headerTitle;
+
+  const NewsListPage({@required this.listEntities, @required this.headerTitle});
 
-class _NewsPageState extends State<NewsListPage> {
   @override
   Widget build(BuildContext context) {
     return Scaffold(
       body: SafeArea(
         child: Column(
           crossAxisAlignment: CrossAxisAlignment.start,
           children: [
-            Container(
-              width: double.maxFinite,
-              height: 80,
-            ),
-            _buildNewsSectionHeader(
-              title: Strings.mostViewed,
-              routeBuilder: null,
-            ),
-            SizedBox(height: _Constants.sectionHeaderPadding),
-            PrimaryNewsListView(
-              primaryNewsListEntities: primaryNewsListEntities.take(5).toList(),
-            ),
-            _buildNewsSectionHeader(
-              title: Strings.newest,
-              routeBuilder: (context) => SecondaryNewsListPage(),
-            ),
+            _buildHeader(context),
             SizedBox(height: _Constants.sectionHeaderPadding),
             SecondaryNewsListView(
```

Comment:
hmm jeżeli klasa nazywa się news_list_page to analogicznie nie wiem czy nie powinien mieć po prostu korzystać z NewsListView - analogicznie jak napisałem wcześniej dot. tego wrzucania mu SecondaryNewsListViewEntity, chyba, że lepiej zmienić nazwę NewsListPage na SecondaryNewsListPage? to już zależy jakie mamy podejście, jeżeli traktujemy to stricte UI to można by to zmienić na SecondaryNewsListPage, jeżeli primaryNews i secondaryNews odnoszą się do innych endpointów, więc stanowią różne obiekty domenowe to skłaniałbym się ku propozycji z którą opisałem w home_page czyli żeby jednak jeżeli mamy news_list_page to nie wrzucać do niego secondary_news_list i secondary_list_view_entity, tylko odpowiednio newsList i newsListViewEntity

Reply:
jw